tx-parser/src/lib.rs

```diff
@@ -175,13 +192,61 @@ impl<'a> Tx<&'a [edn::Value]> {
     }
 }
 
+impl<'a> Tx<&'a [edn::Value]> {
+    pub fn parse_entid_or_lookup_ref_or_temp_id(input: &'a [edn::Value]) -> std::result::Result<EntidOrLookupRefOrTempId, errors::Error> {
+        (Tx::<_>::entid_or_lookup_ref_or_temp_id(), eof())
+            .map(|(es, _)| es)
+            .parse(input)
+            .map(|x| x.0)
+            .map_err::<ValueParseError, _>(|e| e.translate_position(input).into())
+            .map_err(|e| Error::from_kind(ErrorKind::ParseError(e)))
+    }
+}
+
+/// Remove any :db/id value from the given map notation, converting the returned value into
+/// something suitable for the entity position rather than something suitable for a value position.
+///
+/// This is here simply to not expose some of the internal APIs of the tx-parser.
+pub fn remove_db_id(map: &mut MapNotation) -> std::result::Result<Option<EntidOrLookupRefOrTempId>, errors::Error> {
+    // TODO: extract lazy defined constant.
+    let db_id_key = Entid::Ident(edn::NamespacedKeyword::new("db", "id"));
+
+    let db_id: Option<EntidOrLookupRefOrTempId> = if let Some(id) = map.remove(&db_id_key) {
```

Comment:
Prefer `Option::map` to `if let` here, I think — that `else None` at the end caused a "huh" and a skim back.